### PR TITLE
ReactWeek3Homework

### PR DESCRIPTION
**Pull request by CharlesKimMaina**

### CODE REVIEW

File: react/week3/todoform/src/App.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react";
+import Timer from "./Timer";
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      input: "",
```
Comment (benna100) [inline react/week3/todoform/src/App.js:8]:
```suggestion
      inputValue: "",
```

File: react/week3/todoform/src/App.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      date: "",
+      time: 0,
+      todolist: [],
```
Comment (benna100) [inline react/week3/todoform/src/App.js:11]:
```suggestion
      todoList: [],
```

File: react/week3/todoform/src/App.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          onChange={this.changeHandle}
+          value={this.state.input}
+          placeholder="test item"
+        />
+        <br />
+        deadline{" "}
+        <input type="date" value={this.state.date} onChange={this.changeDate} />
+        <br />
+        <button onClick={this.addTodo}>Add Todo</button>
+        <ul>
```
Comment (benna100) [inline react/week3/todoform/src/App.js:31]: This could be a TodoList component with the ul and li

File: react/week3/todoform/src/App.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  deleteTodo = (event) => {
+    const index = event.target.dataset.index;
+    this.setState((state) => {
+      const todolist = [...state.todolist];
+      todolist.splice(index, 1);
+      return { todolist: todolist };
+    });
+  };
```
Comment (benna100) [inline react/week3/todoform/src/App.js:54]: NIce!

File: react/week3/todoform/src/App.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+  changeHandle = (event) => {
```
Comment (benna100) [inline react/week3/todoform/src/App.js:56]: Vague function name, im not sure what this means. Maybe `handleInputValue` is better?

File: react/week3/todoform/src/App.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  };
+
+  addTodo = () => {
+    this.setState((state) => ({
+      todolist: [...state.todolist, state.input],
+      input: "",
+    }));
+  };
+
+  changeDate = (event) => {
```
Comment (benna100) [inline react/week3/todoform/src/App.js:69]: This is good!

File: react/week3/todoform/src/Timer.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+class Timer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      elapsedTime: null,
+    };
+
+    this.countUp = this.countUp.bind(this);
+    this.startCounting = this.startCounting.bind(this);
```
Comment (benna100) [inline react/week3/todoform/src/Timer.js:11]: Do this in componentDidMount

File: react/week3/todoform/src/Timer.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  startCounting() {
+    setInterval(this.countUp, 2000);
+  }
+
+  countUp() {
+    this.setState(({ elapsedTime }) => ({ elapsedTime: elapsedTime + 1 }));
+  }
```
Comment (benna100) [inline react/week3/todoform/src/Timer.js:20]: NIce!

**Review by benna100 — COMMENTED** [review]
Good job man! Looks good overall. To the point.

Hope yopu can use my comments

File: react/week3/todoform/src/App.js (hunk 5 above)
Comment (CharlesKimMaina) [reply react/week3/todoform/src/App.js:56]: yes true, changeHandle is a bit vague ,.,.,I have changed it to HandleInputValue,.Thanks

File: react/week3/todoform/src/Timer.js (hunk 7 above)
Comment (CharlesKimMaina) [reply react/week3/todoform/src/Timer.js:11]: yes great

**Pull request merged**